lib/util/git/index.ts

```diff
@@ -42,6 +43,50 @@ export { setPrivateKey } from './private-key';
 // TODO: fix upstream types https://github.com/steveukx/git-js/issues/704
 const ResetMode = (simpleGit.default as any).ResetMode as typeof _ResetMode;
 
+// Retry parameters
+const retryCount = 5;
+const delaySeconds = 3;
+const delayFactor = 2;
+
+// A generic wrapper for simpleGit.* calls to make them more fault-tolerant
+export async function gitRetry<T>(gitFunc: () => Promise<T>): Promise<T> {
+  let round = 0;
+  while (round <= retryCount) {
+    if (round > 0) {
+      logger.debug({ round }, 'gitRetry round');
+    }
+    try {
+      return await gitFunc();
+    } catch (err) {
+      logger.debug({ err }, `Git function thrown`);
+      try {
+        // Try to transform the Error to ExternalHostError
+        checkForPlatformFailure(err);
+        throw err;
+      } catch (errChecked) {
```

Comment:
What's the reason for needing nested try/catch/throw?

Reply:
I think `checkForPlatformFailure` should return the `Error` object, we will need it later for strict-null-checks anyways. Thes we don't need this extra try-catch